releng/docker/Dockerfile

```diff
@@ -7,34 +9,86 @@
 # repository you want on /zbm inside the container.
 
 # Use the official Void Linux container
-FROM ghcr.io/void-linux/void-glibc-full:latest
+FROM ghcr.io/void-linux/void-glibc-full 
+LABEL org.opencontainers.image.authors="ZFSBootMenu Team, https://zfsbootmenu.org"
+# ZFSBootMenu commit hash, tag or branch name used by
+# default to build ZFSBootMenu images (default: master)
 ARG ZBM_COMMIT_HASH
+# Include the specified Void Linux package in the image
+#
+# (multiple entries must be seperated by spaces)
+ARG PACKAGES=
+ARG XBPS_REPOS="https://repo-fastly.voidlinux.org/current"
+# Include the specified Void Linux kernel series in the image; kernel
+# series take the form <major>.<minor> and correspond to Void packages
+# linux<kver> and linux<kver>-headers
+#
+# Default: install 5.10, 5.15 and 6.1
+#	
+# (multiple entries must be seperated by spaces)
+ARG KERNELS="linux5.10 linux5.15 linux6.1" 
+
+# Note, the following is executed using /bin/sh
+RUN <<-EOF 
+	[ -n "${KERNELS}" ] || { echo "ARG KERNELS must contain a value"; exit 1; }
+	mkdir -p /etc/xbps.d
+	for repo in "${XBPS_REPOS}"; do
+		echo "repository=$repo" >> /etc/xbps.d/00-custom-repos.conf
+	done
+	# Ensure everything is up-to-date
+	xbps-install -Suy xbps && xbps-install -uy bash
+	# Install custom packages
+	if [ -n "${PACKAGES}" ]; then
+		echo "INFO: adding extra Void packages: ${PACKAGES}"
+		xbps-install -y $PACKAGES
+	fi
+	EOF
 
-# Ensure everything is up-to-date
-RUN xbps-install -Suy xbps && xbps-install -uy
+# Switch to bash
+SHELL ["/bin/bash", "-c"]
 
-# Prefer an LTS version over whatever Void thinks is current
-RUN echo "ignorepkg=linux" > /etc/xbps.d/10-nolinux.conf \
-	&& echo "ignorepkg=linux-headers" >> /etc/xbps.d/10-nolinux.conf
+RUN <<-EOF 
+	# Prefer an LTS version over whatever Void thinks is current
+	cat > /etc/xbps.d/10-nolinux.conf <<-CAT
+		ignorepkg=linux
+		ignorepkg=linux-headers
+	CAT
 
-# Install components necessary to build the image
-RUN xbps-query -Rp run_depends zfsbootmenu | xargs xbps-install -y
-RUN xbps-install -y linux5.10 linux5.10-headers \
-	zfs gummiboot-efistub curl yq-go bash kbd \
-	terminus-font dracut mkinitcpio cryptsetup
+	# Prevent initramfs kernel hooks from being installed
+	cat > /etc/xbps.d/15-noinitramfs.conf <<-CAT
+		noextract=/usr/libexec/mkinitcpio/*
+		noextract=/usr/libexec/dracut/*
+	CAT
```

Comment:
Nested heredocs are more difficult to read. Is there a reason this is preferred over `COPY <<-EOF` syntax I see in the documentation?